tools/install.py

```diff
@@ -85,7 +90,7 @@ def npm_files(action):
   # npm has a *lot* of files and it'd be a pain to maintain a fixed list here
   # so we walk its source directory instead...
   for dirname, subdirs, basenames in os.walk('deps/npm', topdown=True):
-    subdirs[:] = filter('test'.__ne__, subdirs) # skip test suites
+    subdirs[:] = [subdir for subdir in subdirs if subdir != 'test']
```

Comment:
Is the __[:]__ needed?

Reply:
@thefourtheye could you put a `TODO` here. I actually think we should not mess with internal `npm` directories at all.

Reply:
> The comment above the `for` loop explains why it is done.

IIUC the comment explains why we `walk`, not why the output is filtered ¯\\_(ツ)_/¯, IMHO `npm` should be a black-box and we should install it as is. But sorry for the off-topic comment.

Reply:
Since ATM the [process for vendoring `npm`](https://github.com/nodejs/node/blob/master/doc/guides/maintaining-npm.md#step-4-extract-and-commit-new-npm) is to extract the release tarball, personally I'd say we should install the whole tree as a black box (So it's equivalent to `npm i -g npm`). 
But IMO we should follow up that discussion in a separate PR.

Reply:
@refack Sure, yeah. We might want to hear from other collaborators as well.